### PR TITLE
Fix 'openssl passwd' with arguments -1 or -apr1

### PR DESCRIPTION
**Pull request by levitte**
[RT#4674](https://rt.openssl.org/Ticket/Display.html?id=4674) issue 1

_login with username 'guest' and password 'guest' if necessary_

### CODE REVIEW

**Comment by levitte** [comment]
As you can see, I also added a test to check that passwd works as promised.

File: apps/passwd.c [hunk 1]
```diff
@@ -319,6 +319,7 @@ static char *md5crypt(const char *passwd, const char *magic, const char *salt)
         || !EVP_DigestUpdate(md, magic, magic_len)
         || !EVP_DigestUpdate(md, "$", 1)
         || !EVP_DigestUpdate(md, salt_out, salt_len))
+        goto err;
```
Comment (richsalz) [inline apps/passwd.c]: OMG.  Our very own goto fail! :)

**Comment by richsalz** [comment]
for master and 1.1.0 i assume.
+1

File: apps/passwd.c (hunk 1 above)
Comment (levitte) [inline apps/passwd.c]: Yup.  I felt physical pain when discovering it...

**Comment by levitte** [comment]
> for master and 1.1.0 i assume.

Yup.  See the lables

**Comment by levitte** [comment]
Merged

**Pull request closed without merging**